src/registry/search.rs

```diff
@@ -17,7 +44,358 @@ pub struct RegistrySearchArgs {
     #[arg(long, default_value = "false")]
     lineage: bool,
 
-    /// The telemetry schema containing the versions (url or file)
-    #[arg(short, long)]
-    pub schema: Option<String>,
+    /// Parameters to specify the diagnostic format.
+    #[command(flatten)]
+    pub diagnostic: DiagnosticArgs,
+
+    /// An (optional) search string to use.  If specified, will return matching values on the command line.
+    /// Otherwise, runs an interactive terminal UI.
+    pub search_string: Option<String>,
+}
+
+#[derive(thiserror::Error, Debug, serde::Serialize, Diagnostic)]
+enum Error {
+    #[error("{0}")]
+    StdIoError(String),
+}
+
+impl From<std::io::Error> for Error {
+    fn from(e: std::io::Error) -> Self {
+        Error::StdIoError(e.to_string())
+    }
+}
+
+// Our search application state.
+// This will be updated by the `process` method, which handles events.
+//
+// It is then used to render in the `render` method, which we try to keep stateless.
+struct SearchApp<'a> {
+    // The current resolved schema that we should be searching.
+    schema: &'a ResolvedTelemetrySchema,
+    // A text-input area where users can enter a search string.
+    search_area: TextArea<'a>,
+    // The current selected index in search results.  Need to be manually cleared when new search strings are entered.
+    selected_result_index: Option<usize>,
+}
+
+impl<'a> SearchApp<'a> {
+    // Creates a new search application for a given resolved schema.
+    fn new(schema: &'a ResolvedTelemetrySchema) -> SearchApp<'a> {
+        let mut search_area = TextArea::default();
+        search_area.set_placeholder_text("Enter search string");
+        search_area.set_block(
+            Block::default()
+                .borders(Borders::TOP)
+                .border_style(Style::default().fg(Color::Gray))
+                .title("Search (press `Esc` or `Ctrl-Q` to stop running) ")
+                .title_style(Style::default().fg(Color::Green)),
+        );
+        SearchApp {
+            schema,
+            search_area,
+            selected_result_index: None,
+        }
+    }
+
+    // Renders the title component of the UI.
+    fn title(&self) -> Paragraph<'a> {
+        let title_block = Block::default()
+            .borders(Borders::TOP)
+            .style(Style::default().bg(Color::Black))
+            .border_style(Style::default().fg(Color::Gray))
+            .title_alignment(ratatui::layout::Alignment::Center)
+            .title_style(Style::default().fg(Color::Green))
+            .title("Weaver Search");
+        let title_contents = Line::from(vec![Span::styled(
+            format!(
+                "Loaded {0:?} registries w/ {1} attributes",
```

Comment:
Is the registry the entire set of groups starting with registry. or is each group starting with registry. considered an individual registry? If as I think it's the former then message should be something like "Loaded {0:?} groups w/ {1} attributes".